src/braket/aws/aws_device.py

```diff
@@ -274,6 +275,79 @@ def arn(self) -> str:
         """str: Return the ARN of the device"""
         return self._arn
 
+    @property
+    def is_available(self) -> bool:
+        """bool: Return if the device is currently available"""
+        if self.status != "ONLINE":
+            return False
+
+        current_datetime_utc = datetime.utcnow()
+        for execution_window in self.properties.service.executionWindows:
+            weekday = current_datetime_utc.weekday()
+            current_time_utc = current_datetime_utc.time()
+
+            if (
+                execution_window.windowEndHour < execution_window.windowStartHour
+                and current_time_utc < execution_window.windowEndHour
+            ):
+                weekday = weekday - 1
+                if weekday == -1:
+                    weekday = 6
```

Comment:
Might be possible to reduce this to `weekday = (weekday - 1) % 7`

Reply:
Updated as suggested.